src/flashbar/__tests__/runtime-action.test.tsx

```diff
@@ -178,6 +178,101 @@ test('propagates flash message context into callback', async () => {
   });
 });
 
+test('propagates state changes to update callback', async () => {
+  const updateContentSpy = jest.fn();
+  const testAction: ActionConfig = {
+    ...defaultAction,
+    updateContent: (container, context) => updateContentSpy(context.contentRef.current?.textContent),
+  };
+
+  awsuiPlugins.flashbar.registerAction(testAction);
+  const { rerender } = render(<Flashbar items={[{ ...defaultItem, content: 'Initial content' }]} />);
+  await delay();
+  expect(updateContentSpy).toHaveBeenCalledTimes(0);
+  rerender(<Flashbar items={[{ ...defaultItem, content: 'Updated content' }]} />);
+  expect(updateContentSpy).toHaveBeenCalledTimes(1);
+  expect(updateContentSpy).toHaveBeenCalledWith('Updated content');
+});
+
+test('container reference is permanent between mount/update/unmount', async () => {
+  let container: HTMLElement | null = null;
+  const testAction: ActionConfig = {
+    id: 'test-action',
+    mountContent: jest.fn(newContainer => (container = newContainer)),
+    updateContent: jest.fn(newContainer => expect(container).toBe(newContainer)),
+    unmountContent: jest.fn(newContainer => expect(container).toBe(newContainer)),
+  };
+  awsuiPlugins.flashbar.registerAction(testAction);
+  const { rerender } = render(<Flashbar items={[{ ...defaultItem }]} />);
+  await delay();
+  expect(testAction.mountContent).toHaveBeenCalledTimes(1);
+  expect(testAction.updateContent).toHaveBeenCalledTimes(0);
+  expect(testAction.unmountContent).toHaveBeenCalledTimes(0);
+  rerender(<Flashbar items={[{ ...defaultItem }]} />);
+  expect(testAction.mountContent).toHaveBeenCalledTimes(1);
+  expect(testAction.updateContent).toHaveBeenCalledTimes(1);
+  expect(testAction.unmountContent).toHaveBeenCalledTimes(0);
+  rerender(<></>);
+  expect(testAction.mountContent).toHaveBeenCalledTimes(1);
+  expect(testAction.updateContent).toHaveBeenCalledTimes(1);
+  expect(testAction.unmountContent).toHaveBeenCalledTimes(1);
+});
+
+test('allows conditionally render content when an item changes', async () => {
```

Comment:
This unit test may seem a little long, but it showcases the real example how this API will be used. So I decided to create it this way.